### ISSUE TITLE
Usability: Snap node objects to containing or adjacent node bounding box

### ISSUE THREAD

**Issue by inolasco** [spritebuilder/SpriteBuilder#209]
If I want to set a sprite at the very bottom of a Scene, or if I try to stack a sprite on top of another one, I have to do it manually by moving them carefully.

The editor should be able to snap objects to others based on their bounding box when moving them.

**Comment by jtwigg** [comment]
An interesting idea. 
Additionally you could use a CCLayoutBox but that may be too limiting.
Then alignment menu options might help as well.

**Comment by inolasco** [comment]
It would help, but I think in general when putting sprites in a scene, such as a ground sprite at the very bottom, or laying out stacked blocks, the process could be simpler if they snapped to the edges of other objects, without the need to add more nodes to the scene.

I thought of it while following this tutorial, if it helps illustrate the idea:
https://www.makegameswith.us/tutorials/getting-started-with-spritebuilder/levels/

Had issues getting the ice blocks correctly aligned and stacked :P

**Comment by Ben-G** [comment]
Yeah, would be super super helpful. Snapping would be great. Also the option to select multiple elements an make the edges connect, etc. With some improvements to the editor, SpriteBuilder can be used as a great Level Editor. 

Without it I had a hard time at this weekends Global Games Jam convincing people not to switch to other tools ;)

**Comment by cocojoe** [comment]
Yeah agree that snapping would be a nice feature to look at adding, in the
past I've kept it sensible with powers of 2 objects and sensible grid size
however you can't assume this.

On 27 January 2014 02:45, Ben-G notifications@github.com wrote:

> Yeah, would be super super helpful. Snapping would be great. Also the
> option to select multiple elements an make the edges connect, etc. With
> some improvements to the editor, SpriteBuilder can be used as a great Level
> Editor.
> 
> Without it I had a hard time at this weekends Global Games Jam convincing
> people not to switch to other tools ;)
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/apportable/SpriteBuilder/issues/209#issuecomment-33338956
> .

## 

Martin

**Comment by NoorFatima** [comment]
This issue is fixed.
Device: Mac OS X 10.9.5
SB Version: 1.4.7
